### DISCUSSION TITLE
Cluster transaction on documents with time series

### DISCUSSION THREAD

**Discussion by CMircea** [ravendb/ravendb#12962] [Q&A] [answered]
Hello!

Why are cluster transactions with changes to documents that also have time series (but without any change to those in the transaction) [not supported](https://github.com/ravendb/ravendb/blob/e84bba87d128ef53db8e20866515dea5a6f42a1a/src/Raven.Server/Documents/Handlers/BatchHandler.cs#L178)?

**Comment by ayende** [comment, marked as answer]
Because you may create a new time series in the document. At which point you'll update the document itself.
But a cluster transaction would override that operation, which will lead to a missing time series entry.

Better to create a separate physical document that is the same logical one.

`users/1-A` - modified with cluster transactions
`users/1-A/stats` - has time series

**Reply by CMircea** [reply]
The concern is with regular transactions with added attachments / counters / time series being overwritten by a cluster transaction? That makes sense; though I'd prefer if that would have been configurable, such as a boolean under the .Advanced of the session or something similar. Give me the option to shoot myself in the foot :)

**Reply by ayende** [reply]
Yes, that is the issue, and separating things as documents means that you won't need to shoot yourself at all. The problem with foot guns is that they generate issues, and support calls. Better to have a safe scenario, instead.

**Reply by CMircea** [reply]
Makes sense, I'll use dedicated documents for counters and time series. Thanks for the help!
